### ISSUE TITLE
[upgrade test failure] Initializers should be invisible to controllers by default

### ISSUE THREAD

**Issue by ericchiang** [kubernetes/kubernetes#52585]
cc @kubernetes/sig-api-machinery-test-failures 
cc @kubernetes/kubernetes-release-managers 
/kind bug
/king upgrade-test-failure

Observed on the upgrade cluster test suite

https://k8s-testgrid.appspot.com/master-upgrade#gce-1.7-master-upgrade-cluster-parallel

Example failure

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-new-master-upgrade-cluster-parallel/568

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/extension/initializers.go:108
Expected error:
    <*errors.StatusError | 0xc4202f7680>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {SelfLink: "", ResourceVersion: ""},
            Status: "Failure",
            Message: "Pod \"uninitialized-pod\" is invalid: metadata.initializers.pending[0].name: Invalid value: \"Test\": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')",
            Reason: "Invalid",
            Details: {
                Name: "uninitialized-pod",
                Group: "",
                Kind: "Pod",
                UID: "",
                Causes: [
                    {
                        Type: "FieldValueInvalid",
                        Message: "Invalid value: \"Test\": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')",
                        Field: "metadata.initializers.pending[0].name",
                    },
                ],
                RetryAfterSeconds: 0,
            },
            Code: 422,
        },
    }
    Pod "uninitialized-pod" is invalid: metadata.initializers.pending[0].name: Invalid value: "Test": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/extension/initializers.go:54
				
```

**Comment by liggitt** [comment]
Initializer name validation tightened in 1.8 (and is release noted as needing action), test should be updated with a valid name

**Comment by liggitt** [comment]
should be fixed by https://github.com/kubernetes/kubernetes/pull/52592

**Comment by ericchiang** [comment]
Fix merged into 1.7 branch. Closing.
